tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/AsyncIOServletTest.java

```diff
@@ -431,7 +428,19 @@ public void onWritePossible() throws IOException
 
                         // Wait for the failure to arrive to
                         // the server while we are about to write.
-                        sleep(2000);
+                        try
+                        {
+                            await().atMost(5, TimeUnit.SECONDS).until(() ->
+                            {
+                                out.write(new byte[0]);
```

Comment:
Why the `write()`?

Reply:
* cast to `HttpOutput` and call a direct API that does not aggregate to write some content
* wait until a write fails -- the until() loop should be `write()+isReady()` and check for some "failed" state (because write() does not throw)
* data to write can just be an array of zeroes
* remove `write()` outside until() loop.